lib/index.js

```diff
@@ -80,6 +86,13 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId;
```

Comment:
Hey @mikeng13,

> how is the userId in track.userId() set? I just want to make sure the developer can explicitly set it as a non anonymous event before calling track event

Basically, if you call `.identify()` anywhere before a `.track()` call, we store the `userId` in your cookie and all subsequent `analytics.js` calls like `.page()`, `.track()`, etc. will all have the `userId` automatically attached to the event.

> I called .identify() prior to calling track but the userId was undefined.

Did you pass a `userId`? `.identify('some-id-123')`

> we are able to assign some sort of visitor id to the user for traffic splitting and bucketing purposes. To that end, we need to make that visitor id available here in the tracking call so we can bucket accordingly and determine whether to track a conversion or not. This doesn't necessarily mean that the user is logged in, but it could just mean that we have assigned this visitor and id and we are traffic splitting using this id.

Hmm so you guys have your own concept of `anonymousId`? Does your `optimizelyClientInstance` have an API that gives you read access to that value? Because maybe you could do

```
var userId = track.userId() || optimizelyClientInstance.whatever();
if (userId) { ... }
```

@sperand-io @tsholmes since I'll be OOO will cc you guys here in case I don't get to this.